util/math_reference.h

```diff
@@ -15,22 +15,289 @@
 
 namespace reference {
 /* two argument relational reference */
-int isequal(float x, float y);
-int isnotequal(float x, float y);
-int isgreater(float x, float y);
-int isgreaterequal(float x, float y);
-int isless(float x, float y);
-int islessequal(float x, float y);
-int islessgreater(float x, float y);
-int isordered(float x, float y);
-int isunordered(float x, float y);
+int32_t isequal(float x, float y);
+int64_t isequal(double x, double y);
+template <int N>
+cl::sycl::vec<int32_t, N> isequal(cl::sycl::vec<float, N> a,
+                                  cl::sycl::vec<float, N> b) {
+  return sycl_cts::math::run_rel_func_on_vector<int32_t, float, N>(
+      [](float x, float y) { return isequal(x, y); }, a, b);
+}
+template <int N>
+cl::sycl::vec<int64_t, N> isequal(cl::sycl::vec<double, N> a,
+                                  cl::sycl::vec<double, N> b) {
+  return sycl_cts::math::run_rel_func_on_vector<int64_t, double, N>(
+      [](double x, double y) { return isequal(x, y); }, a, b);
+}
+
+int32_t isnotequal(float x, float y);
```

Comment:
In theory all these could be metaprogrammed as
```c++
template <typename T>
auto isnotequal(const T& a, const T& b) {
  return some_generic_dispatcher<std::not_equal_to>(a, b);
}
```
and writing `some_generic_dispatcher` looks like a good coding exercise. :-)